Add blogpost about v0 recipe changes (#190)
conda/conda-dot-org: blog/2024-11-15-v0-format-modifications.mdx

authors: [kenodegard, chatgpt]
tags: [announcement, conda, conda-build, recipe, cep-13, cep-14]
description: >
  The recent approval of the new recipe format (`recipe.yaml`, v1) does not mean the end of the
  old format (`meta.yaml`, v0). The conda community must balance innovation with support for
  legacy systems.
image: img/blog/2024-11-15-v0-format-modifications/banner.png
---

[cep-13]: https://github.com/conda/ceps/blob/main/cep-13.md
[cep-14]: https://github.com/conda/ceps/blob/main/cep-14.md
[rattler-build]: https://prefix-dev.github.io/rattler-build/latest/
[ceps]: https://github.com/conda/ceps
[rattler-recipe]: https://prefix-dev.github.io/rattler-build/latest/reference/recipe_file/
[rattler-migration]: https://prefix-dev.github.io/rattler-build/latest/converting_from_conda_build/
[crm]: https://conda-incubator.github.io/conda-recipe-manager/index.html

The recent approval of a new recipe format (v1) by the Conda Steering Council
via [CEP-13][cep-13] and [CEP-14][cep-14] marks an important milestone in the
evolution of package management in the conda ecosystem. While the community
embraces the potential improvements that come with the new format, we

> acknowledge that the v1 recipe format is still under development and is
> being prototyped in [rattler-build][rattler-build]. Consequently, the v0 recipe

Reviewer: ```suggestion
acknowledge that the v1 recipe format is still undergoing development
in [rattler-build][rattler-build]. Consequently, the v0 recipe
```

I think this is what Wolf meant.

format remains widely in use, emphasizing the need to balance innovation with support
for legacy systems.

<!-- truncate -->

While the focus is on innovating with the v1 format, the transition and adoption
will take time. In the meantime, we still need to remain responsive to the challenges
and requirements of those who continue to rely on the v0 format for their packaging
needs. Updates to the v0 format should be driven by the needs of the user
community, guided by user feedback, and aimed at addressing pain points.

To initiate changes to the v0 conda-build recipe format, the following workflow
is followed:

1. **Conda Enhancement Proposal**: Any proposed modifications to the v0 format should
   also be proposed in the v1 format. The rationale being, if it's necessary for v0

>    it should also be necessary for v1. All proposals go through the CEP process,

Reviewer 2: ```suggestion
   it should also be necessary for v1. All proposals go through the CEP process;
```
